include/rsp.inc

```diff
@@ -263,44 +263,54 @@ makeNotImplemented tnei
     .endm
 .endm
 
-.macro makeLoadInstruction name, opcode
+.macro makeLoadInstruction name, opcode, offsetshift
     .macro \name vt, element, offset, base
         .ifgt (\element >> 4)
             .error "Invalid element"
         .else
-            .ifge (\offset)
-                .ifgt (\offset - 63)
-                    .error "Invalid offset - valid range: [-64, 63]"
-                .else
-                    loadVector \opcode, \vt, \element, \offset, \base
-                .endif
+            .ifne ((\offset) % (1 << \offsetshift))
+                .error "Invalid offset - must be multiple of operand size"
             .else
-                .iflt (\offset + 64)
-                    .error "Invalid offset - valid range: [-64, 63]"
+                .ifge (\offset)
+                    .ifgt ((\offset) - (63 << \offsetshift))
+                        .error "Invalid offset - valid range: [-64, 63]"
```

Comment:
This ranges are not the correct ones anymore, so the error message needs to be updated. Can you use variables in the `.error` lines? If not, I think we should resort to duplicate these macros for all different shift values. I know it's verbose but it's code for a greater good: we write lots of duplicated code here to provide a very good assembly support for microcode writers.

Reply:
I didn't find a way to adjust the error messages unfortunately, so I just ended up duplicating the macro.